internal/service/alarms/api/server.go

```diff
@@ -41,6 +44,14 @@ type AlarmsServer struct {
 	AlarmsRepository *repo.AlarmsRepository
 	// ClusterServer contains the cluster server client and fetched objects
 	ClusterServer *clusterserver.ClusterServer
+	// Wg to allow alarm server level background tasks to finish before graceful exit
+	Wg sync.WaitGroup
+	// NotificationProvider to handle new events
+	NotificationProvider notifier.NotificationProvider
```

Comment:
Are the last 3 necessary in the `AlarmsServer` object or can they live as locals in the start function?

Reply:
Removed Wg since now we also check the notification during restart...so no need to wait. 
Removed SubscriptionProvider since its not used. 

But keeping Notifier to call Notify + NotificationProvider since its used to grab the notifications with  GetNotifications (before Notify)

Reply:
On second thought...brought back the server Wg. Hopefully its going to be basically noop - but if we ever have a something stuck in notify during server exit will exit with error. wdyt?